test/PauseTest.php

```diff
@@ -0,0 +1,131 @@
+<?php
+namespace Resque\Plugins\Tests;
+
+use Credis_Client;
+use Resque;
+use Resque_Job_DontCreate;
+use Resque\Plugins\Pause;
+use PHPUnit_Framework_TestCase;
+
+/**
+ * Pause tests.
+ *
+ * @package     PHP Resque Pause
+ * @author      Wedy Chainy <askwedi@wedipedia.org>
+ * @license     http://www.opensource.org/licenses/mit-license.php
+ */
+class PauseTest extends PHPUnit_Framework_TestCase
+{
+    protected $resque;
+    protected $redis;
+
+    public function setUp()
+    {
+        $config = file_get_contents(REDIS_CONF);
+        preg_match('#^\s*port\s+([0-9]+)#m', $config, $matches);
+        $this->redis = new Credis_Client('localhost', $matches[1]);
+        $this->redis->flushAll();
+
+    }
+
+    public function testPause()
+    {
+        Resque::enqueue('upgrade:test', 'test');
+        Resque::enqueue('upgrade:test', 'test');
+        $this->assertEquals(2, Resque::size('upgrade:test'));
+        $this->assertTrue((bool)Pause::pause('upgrade:test'));
+        $this->assertEquals(0, Resque::size('upgrade:test'));
+    }
+
+    public function testPauseNonExistQueue()
+    {
+        $this->assertFalse((bool)Pause::pause('upgrade:test2'));
+    }
+
+    public function testUnpause()
+    {
+        Resque::enqueue('upgrade:test3', 'test');
+        Resque::enqueue('upgrade:test3', 'test');
+        Pause::pause('upgrade:test3');
+        $this->assertTrue((bool)Pause::unpause('upgrade:test3'));
+        $this->assertEquals(2, Resque::size('upgrade:test3'));
+    }
+
+    public function testUnpauseButNeverBeenPausedBefore()
+    {
+        Resque::enqueue('upgrade:test4', 'test');
+        Resque::enqueue('upgrade:test4', 'test');
+        $this->assertFalse((bool)Pause::unpause('upgrade:test4'));
+    }
+
+    public function testIsPaused()
+    {
+        Resque::enqueue('upgrade:test5', 'test');
+        Pause::pause('upgrade:test5');
+        $this->assertTrue((bool)Pause::isPaused('upgrade:test5'));
+    }
+
+    public function testIsPaused2()
+    {
+        Resque::enqueue('upgrade:test6', 'test');
+        $this->assertFalse((bool)Pause::isPaused('upgrade:test6'));
+    }
+
+    public function testIsPaused3()
+    {
+        Resque::enqueue('upgrade:test7', 'test');
+        Pause::pause('upgrade:test7');
+        Pause::unpause('upgrade:test7');
+        $this->assertFalse((bool)Pause::isPaused('upgrade:test7'));
+    }
+
+    public function testPauseCallback()
+    {
+        # we have 2 in queue:upgrade:test8
+        Resque::enqueue('upgrade:test8', 'test');
+        Resque::enqueue('upgrade:test8', 'test');
+
+        # pause it!
+        Pause::pause('upgrade:test8');
+
+        try {
+            #  before enqueue pauseCallback
+            Pause::pauseCallback('test', array(), 'upgrade:test8', 'dummyid');
+        } catch (Resque_Job_DontCreate $e) {
```

Comment:
Could you give a bit more explanation as to why you're catching this? I'd like to see an assert (above, in the try) that describes why it's OK to ignore this.